src/main/scala/scala/collection/immutable/next/package.scala

```diff
@@ -0,0 +1,51 @@
+/*
+ * Scala (https://www.scala-lang.org)
+ *
+ * Copyright EPFL and Lightbend, Inc.
+ *
+ * Licensed under Apache License 2.0
+ * (http://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package scala.collection.immutable
+
+package object next {
+
+  implicit class NextLazyListExtensions[T](private val ll: LazyList[T]) extends AnyVal {
+    /**
+     * When called on a finite `LazyList`, returns a circular structure
+     * that endlessly repeats the elements in the input.
+     * The result is a true cycle occupying only constant memory.
+     *
+     * Does not force the input list (not even its empty-or-not status).
+     *
+     * Safe to call on unbounded input, but in that case the result is not a cycle
+     * (not even if the input was).
+     *
+     * Note that some `LazyList` methods preserve cyclicality and others do not.
+     * So for example the `tail` of a cycle is still a cycle, but `map` and `filter`
+     * on a cycle do not return cycles.
+     */
+    def cycle: LazyList[T] =
+      // case 1: the input is already known to be empty
+      // (the test can be changed to ll.knownIsEmpty when this code moves to stdlib)
+      if (ll.knownSize == 0) LazyList.empty
+      // we don't want to force the input's empty-or-not status until we must.
+      // `LazyList.empty #:::` accomplishes that delay
+      else LazyList.empty #::: {
+        // case 2: the input is later discovered to be empty
+        // (distinguishing this from case 3 isn't necessary for correctness, but we
```

Comment:
this is not the case; if you allowed an empty `LazyList` through to case 3, this would happen:

```scala
scala> val ll = LazyList.empty[Int]
val ll: scala.collection.immutable.LazyList[Int] = LazyList()

scala> lazy val result: LazyList[Int] = ll #::: result
lazy val result: LazyList[Int] // unevaluated

scala> result.isEmpty
java.lang.RuntimeException: self-referential LazyList or a derivation thereof has no more elements
  at scala.collection.immutable.LazyList.scala$collection$immutable$LazyList$$state$lzycompute(LazyList.scala:256)
  at scala.collection.immutable.LazyList.scala$collection$immutable$LazyList$$state(LazyList.scala:252)
  at scala.collection.immutable.LazyList.$anonfun$lazyAppendedAll$1(LazyList.scala:379)
  at scala.collection.immutable.LazyList.scala$collection$immutable$LazyList$$state$lzycompute(LazyList.scala:259)
  at scala.collection.immutable.LazyList.scala$collection$immutable$LazyList$$state(LazyList.scala:252)
  at scala.collection.immutable.LazyList.isEmpty(LazyList.scala:269)
  ... 32 elided
```

Reply:
it would be kind of as if you tried to have an infinitely long `ll #::: ll #::: ll #::: ll #::: ll #::: // ...`